LEGO1/decomp.h

```diff
@@ -0,0 +1,10 @@
+#ifndef DECOMP_H
+#define DECOMP_H
+
+#define DECOMP_STATIC_ASSERT(V) namespace { typedef int foo[(V)?1:-1]; }
+
+typedef unsigned char undefined;
+typedef unsigned short undefined2;
+typedef unsigned int undefined4;
```

Comment:
I'm not sure these are necessary, couldn't we just use `MxU32`, `MxU16`, etc. from `mxtypes.h`?

Reply:
The idea is to use undefinedX where we don't know what type something has. It can be a value, or a pointer.
`MxU32` are used when you are 100% sure it's a 32-bit integer.

Reply:
The final goal is of this project is to only have `DECOMP_STATIC_ASSERT`'s at the end.

Reply:
You can see it as self-documenting code.

Reply:
Sure, but if it's called `m_unk50`, doesn't that already indicate that we don't know what it is?

Reply:
Yes, and these members are also used in functions.
A simple grep for `undefined` will show what is currenty unknown. MxUxx is fuzzy.

Reply:
Generally I think it's a good idea. Just like with other measures too though, I think it's better - in case we decide for this - that we don't gradually introduce it but ideally apply it everywhere in the code base at once and establish its usage as a "style" rule. 

Maybe the same should be done with the types from `mxtypes.h` - currently it's used in *some* places, but not everywhere it probably should be. We could also define a rule where it should be used. I always find consistency, even in those small things, to be extremely helpful.

With regards to the naming of variables: generally there are always two possible "unknown" components I suppose, the type and the "purpose" (designated by its name). It's possible that both the type and the purpose are unknown. Sometimes however we know the type, but not its purpose. So I can imagine 4 cases like this:

```
undefined4 m_unk4;
undefined4 m_length; // probably relatively rare
MxU32 m_unk4;
MxU32 m_length;
```

By introducing something like the `undefined*` types we can document that we don't know which type it is (yet). I think that would be good

Reply:
Yeah okay, seeing it in use (e.g. writing `undefined[0x16c]` to pad a struct instead of `char[0x16c]`) has convinced me it probably is helpful.

I agree it should probably be made a rule, along with using the `mxtypes`. The exports have a lot of `unsigned char`s that almost certainly resolve to MxBool but don't yet in our code.